osu.Game/Overlays/ChatConsole.cs

```diff
@@ -0,0 +1,182 @@
+                    Colour = new Color4(0.1f, 0.1f, 0.1f, 0.4f)
+                },
+                content = new Container
+                {
+                    RelativeSizeAxes = Axes.Both,
+                }
+            };
+        }
+
+        public override void Load()
+        {
+            base.Load();
+            initializeChannels();
+        }
+
+        private long? lastMessageId;
+
+        List<Channel> careChannels;
+
+        private void initializeChannels()
+        {
+            careChannels = new List<Channel>();
+
+            //if (api.State != APIAccess.APIState.Online)
+            //  return;
+
+            Add(new FlowContainer
+            {
+                RelativeSizeAxes = Axes.Both,
+                Direction = FlowDirection.VerticalOnly
+            });
+
+            SpriteText loading;
+            Add(loading = new SpriteText
+            {
+                Text = @"Loading available channels...",
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                TextSize = 40,
+            });
+
+            messageRequest?.Cancel();
+
+            ListChannelsRequest req = new ListChannelsRequest();
+            req.Success += delegate (List<Channel> channels)
+            {
+                Scheduler.Add(delegate
+                {
+                    loading.FadeOut(100);
+                });
+
+                addChannel(channels.Find(c => c.Name == @"#osu"));
+                //addChannel(channels.Find(c => c.Name == @"#lobby"));
+                //addChannel(channels.Find(c => c.Name == @"#english"));
+
+                messageRequest = Scheduler.AddDelayed(() => FetchNewMessages(api), 1000, true);
+            };
+            api.Queue(req);
+        }
+
+        private void addChannel(Channel channel)
+        {
+            Add(channelDisplay = new ChannelDisplay(channel));
+            careChannels.Add(channel);
+        }
+
+        GetMessagesRequest fetchReq;
+
+        public void FetchNewMessages(APIAccess api)
+        {
+            if (fetchReq != null) return;
+
+            fetchReq = new GetMessagesRequest(careChannels, lastMessageId);
+            fetchReq.Success += delegate (List<Message> messages)
+            {
+                foreach (Message m in messages)
+                {
+                    careChannels.Find(c => c.Id == m.ChannelId).AddNewMessages(m);
+                }
+
+                lastMessageId = messages.LastOrDefault()?.Id ?? lastMessageId;
+
+                Debug.Write("success!");
+                fetchReq = null;
+            };
+            fetchReq.Failure += delegate
+            {
+                Debug.Write("failure!");
+                fetchReq = null;
+            };
+
+            api.Queue(fetchReq);
+        }
+
+        private ChatConsoleState state;
+
+        public ChatConsoleState State
+        {
+            get
+            {
+                return state;
+            }
+

```

Comment:
Style nit - just join all these lines together and drop the excess whitespace/newlines.